manifests/config.pp

```diff
@@ -44,7 +44,15 @@
 
   case $ensure {
     'present': {
-      $changes = "set ${name} '${value}'"
+      if $comment {
+        $changes = [
+          "set ${name} '${value}'",
+          "insert #comment before $name",
```

Comment:
I do think that managing this idempotently will probably require a type/provider. Are you familiar with augeasproviders.com?